src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,278 @@
+                            'warnings', 'wave', 'weakref', 'webbrowser', 'wsgiref', 'xdrlib', 'xml', 'xmlrpc',
+                            'zipapp', 'zipfile', 'zipimport', 'zoneinfo', '_aix_support', '_bootlocale',
+                            '_bootsubprocess', '_collections_abc', '_compat_pickle', '_compression', '_markupbase',
+                            '_osx_support', '_pydecimal', '_pyio', '_py_abc', '_sitebuiltins', '_strptime',
+                            '_threading_local', '_weakrefset', '__future__', '__phello__', '__pycache__']
+        standard_lib.extend(sys.builtin_module_names)
+        standard_lib.append("sas")
+
+        for module_name in sys.modules.keys():
+
+            package_name = module_name.split('.')[0]
+
+            # A built in python module or a local file, which have no version, only the python/SasView version
+            if package_name in standard_lib or package_name in package_versions_dict:
+                continue
+
+            # Modules that require specific methods to get the version number
+            if "PyQt5" in module_name:
+                try:
+                    from PyQt5.QtCore import PYQT_VERSION_STR
+                except Exception as e:
+                    # Unable to access PyQt5
+                    err_version_dict[module_name] = f"Unknown: {e} when attempting to get PyQt5 version"
+                    pass
+                else:
+                    package_versions_dict["PyQt"] = PYQT_VERSION_STR
+                try:
+                    from PyQt5.QtCore import QT_VERSION_STR
+                except Exception as e:
+                    # Unable to access Qt
+                    err_version_dict[module_name] = f"Unknown: {e} when attempting to get Qt version"
+                    pass
+                else:
+                    package_versions_dict["Qt"] = QT_VERSION_STR
+                continue
+
+            # Import module
+            try:
+                package = __import__(package_name)
+            except Exception as e:
+                err_version_dict[package_name] = f"Unknown: {e} when attempting to import module"
+                continue
+
+            # Different attempts of retrieving the modules version
+            if hasattr(package, '__version__'):
+                # Module has __version__ attribute
+                try:
+                    package_versions_dict[package_name] = package.__version__
+                except Exception as e:
+                    # Unable to access module
+                    err_version_dict[package_name] = f"Unknown: {e} when attempting to access {package_name} " \
+                                                     f"version using .__version__"
+                    pass
+                continue
+
+            if hasattr(package, '.version'):
+                # Module has .version attribute
+                try:
+                    if isinstance(package.version, str) or isinstance(package.version, float):
```

Comment:
'float' is very unlikely. Do you have a counterexample? A tuple of integers and strings is much more likely.